### PR TITLE
Fix the hidden close button in the drawer.

### PR DESCRIPTION
**Pull request by notlee**
If the dropdown close button at the end of the dropdown menu
is clicked it will not close the menu if the menu has the current
page within it and the drawer is open.

### CODE REVIEW

File: src/js/drop-down.js [hunk 1]
```diff
@@ -45,28 +45,47 @@ class DropDown {
...
+			// Bail if there's no parent menu to toggle. Note: IE11 does not
+			// support `Array.prototype.find()` or `Element.closest()`,
+			// adding a polyfill requirement is a breaking change
+			let parentMenu;
+			for (const navItem of this.navItems) {
+				if (navItem.contains(event.target)) {
+					parentMenu = navItem;
+					break; // found the parent menu
+				}
+			}
```
Comment (chee) [inline src/js/drop-down.js:71]:
```suggestion
			let parentMenu = this.navItems.find(item => item.contains(event.target));
```

File: src/js/drop-down.js [hunk 2]
```diff
@@ -45,28 +45,47 @@ class DropDown {
...
 				this.reset();
 				return;
 			}
 
-			const target = event.target.closest('li');
-			if (!DropDown.isExpanded(target) && event.target.type === 'button') {
+			// Bail if there's no parent menu to toggle. Note: IE11 does not
+			// support `Array.prototype.find()` or `Element.closest()`,
+			// adding a polyfill requirement is a breaking change
+			let parentMenu;
```
Comment (chee) [inline src/js/drop-down.js:65]:
```suggestion
```

File: src/js/drop-down.js (hunk 1 above)
Comment (notlee) [reply src/js/drop-down.js:71]: `find` would require a new polyfill to support IE11, which we can't introduce without a breaking change. :(

File: src/js/drop-down.js (hunk 2 above)
Comment (notlee) [reply src/js/drop-down.js:65]: see above

File: src/js/drop-down.js [hunk 3]
```diff
@@ -59,8 +59,17 @@ class DropDown {
...
+			for (const navItem of this.navItems) {
+				if (navItem.contains(event.target)) {
+					parentMenu = navItem;
+					break; // found the parent menu
+				}
+			}
```
Comment (chee) [inline src/js/drop-down.js:71]:
```suggestion
			let parentMenu;
			for (let itemIndex = 0; itemIndex < this.navItems.length; itemIndex++) {
				const navItem = this.navItems[itemIndex];

				if (navItem.contains(event.target)) {
					parentMenu = navItem;
					break; // found the parent menu
				}
			}
```

File: src/js/drop-down.js (hunk 1 above)
Comment (chee) [reply src/js/drop-down.js:71]: oh i completely missed that ^_^, let me try again

File: src/js/drop-down.js (hunk 3 above)
Comment (notlee) [reply src/js/drop-down.js:71]: Thanks I was actually expecting that `for...of` to be transpilled away, I was just looking into whether that was a reasonable expectation. I think [Babel does](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=GYewTgBAFAxiB2BnALhAlhEwIG0DkAhngDQR4BGeAugJQQDeAUBBHEiADYCmAdByAHMoaGgG5GAXyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=false&presets=env%2Ces2015&prettier=false&targets=&version=7.11.6&externalPlugins=)

File: src/js/drop-down.js [hunk 4]
```diff
@@ -59,8 +59,17 @@ class DropDown {
 				return;
 			}
 
-			// Bail if there's no parent menu to toggle.
-			const parentMenu = event.target.closest('[data-o-header-services-level="1"]');
+			// Bail if there's no parent menu to toggle. Note: IE11 does not
+			// support `Array.prototype.find()` or `Element.closest()`,
+			// adding a polyfill requirement is a breaking change
+			let parentMenu;
+			for (const navItem of this.navItems) {
```
[the remark this answers is not in the document]
Comment (notlee) [reply src/js/drop-down.js:66]: I was actually expecting that to be transpiled away, [see here](https://github.com/Financial-Times/o-header-services/pull/161#discussion_r484443783)

File: src/js/drop-down.js (hunk 3 above)
Comment (chee) [reply src/js/drop-down.js:71]: yeah, if our policy is shipping javascript and expected the consumer to transpile for their platform then i think it is a reasonable expectation.

it would be nice if we could expect users to provide the correct polyfills too, by running something like js-features-analyser on their bundle.

**Review by chee — APPROVED** [review]

File: src/js/drop-down.js (hunk 3 above)
Comment (notlee) [reply src/js/drop-down.js:71]: >yeah, if our policy is shipping javascript and expected the consumer to transpile for their platform then i think it is a reasonable expectation. (and i believe that is our policy)

Just has a look, it is! [JavaScript in components may use ES6 syntax, which product developers are expected to transpile to be ES5 if required](https://origami.ft.com/spec/v1/components/javascript/#feature-stability-and-polyfills). Although we dont specify a tool. Looks like Babel would handle it fine but via [swc](https://github.com/swc-project/swc) an error is thrown so I can't test this in IE as is -- there's [an swc issue here but without discussion](https://github.com/swc-project/swc/issues/1019). Think I'll just commit the`for` loop but wonder if any other components have been caught out by our shift from Babel.

>it would be nice if we could expect users to provide the correct polyfills too, by running something like js-features-analyser on their bundle.

Yeah it would be nice if it was automated in projects for minor/patch releases

**Comment by origamiserviceuser** [comment]
o-header-services bundle size difference from 4.0.12 to b456a54e5809cc54823eee3534daae0c54f8c385
No significant bundle size differences found.

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v4.0.13 :tada:
